crates/turborepo-lsp/src/lib.rs

```diff
@@ -0,0 +1,853 @@
+#![deny(clippy::all)]
+#![warn(clippy::unwrap_used)]
+
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex},
+};
+
+use itertools::Itertools;
+use jsonc_parser::CollectOptions;
+use serde_json::Value;
+use tokio::sync::watch::{Receiver, Sender};
+use tower_lsp::{
+    jsonrpc::{Error, Result as LspResult},
+    lsp_types::*,
+    Client, LanguageServer,
+};
+use turbopath::AbsoluteSystemPathBuf;
+use turborepo_lib::{DaemonClient, DaemonConnector, DaemonPackageDiscovery, DaemonRootHasher};
+use turborepo_repository::{
+    discovery::{self, DiscoveryResponse, PackageDiscovery},
+    package_json::PackageJson,
+};
+
+#[derive(Debug)]
+pub struct Backend {
+    client: Client,
+    repo_root: Arc<Mutex<Option<AbsoluteSystemPathBuf>>>,
+    files: Mutex<HashMap<Url, crop::Rope>>,
+    initializer: Sender<Option<DaemonClient<DaemonConnector>>>,
+    daemon: Receiver<Option<DaemonClient<DaemonConnector>>>,
```

Comment:
We need an Option here because the server does other initialization stuff while the daemon connect future is run to bring down startup times. So to begin with, Daemon will be None after which it can be grabbed at will.